### PR TITLE
Append upstream name using a utility

### PR DESCRIPTION
**Pull request by develohpanda**
This is a minor refactor which should help avoid typos, but more importantly it _contextually connects_ the upstream name to the service host.

It reduces the risk of the service host and upstream name from falling out of sync (even though fixture tests would catch it).

### CODE REVIEW

File: packages/openapi-2-kong/src/declarative-config/upstreams.ts [hunk 1]
```diff
@@ -39,3 +39,5 @@ export function generateUpstreams(api: OpenApi3Spec, tags: string[]) {
 
   return [upstream];
 }
+
+export const appendUpstreamName = (name: string) => `${name}.upstream`;
```
Comment (jackkav) [inline packages/openapi-2-kong/src/declarative-config/upstreams.ts]: I would suggest the wording `appendUpstreamToName`

**Review by jackkav — CHANGES_REQUESTED** [review]

File: packages/openapi-2-kong/src/declarative-config/services.ts [hunk 2]
```diff
@@ -47,7 +48,7 @@ export function generateService(server: OA3Server, api: OpenApi3Spec, tags: stri
     name,
     // remove semicolon i.e. convert `https:` to `https`
     protocol: parsedUrl?.protocol?.substring(0, parsedUrl.protocol.length - 1),
-    host: `${name}.upstream`,
+    host: appendUpstreamName(name),
```
Comment (dimitropoulos) [inline packages/openapi-2-kong/src/declarative-config/services.ts]: apologies if I missed it: is there a test that covers this line?  I was expecting to see one when I saw this PR.  It'd be a test that fails if this line were changed to something like this, for example:
```diff
- host: appendUpstreamName(name),
+ host: name,
```

Comment (dimitropoulos) [reply packages/openapi-2-kong/src/declarative-config/services.ts]: put another way, if we're going to extract this to a utility, it seems prudent to write a test to ensure that the utility is actually used where we expect it to be (in addition to a test that the utility has the behavior we'd expect, which has already been done :smile:).

Comment (develohpanda) [reply packages/openapi-2-kong/src/declarative-config/services.ts]: There are fixture tests

**Review by jackkav — APPROVED** [review]

**Pull request merged**